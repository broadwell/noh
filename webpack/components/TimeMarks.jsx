import React from "react";
import PropTypes from "prop-types";
import { convertSecondsToHhmmss } from "../utils";

const TimeMarks = props => {
  const interval = props.videoDuration / props.numIntervals;
  const marks = [];
  for (let i = 0; i <= props.numIntervals; i += 1) {
    marks.push(
      <div className="time" key={`marker${i * interval}`}>
        {convertSecondsToHhmmss(i * interval)}
      </div>
    );
  }
  return <div className="time-marks">{marks}</div>;
};

TimeMarks.propTypes = {
  videoDuration: PropTypes.number.isRequired,
  numIntervals: PropTypes.number
};

TimeMarks.defaultProps = {
  numIntervals: 6
};

export default TimeMarks;
