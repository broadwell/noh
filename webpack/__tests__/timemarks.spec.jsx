import React from "react";
import { shallow } from "enzyme";
import TimeMarks from "../components/TimeMarks";

describe("<TimeMarks>", () => {
  it("renders as expected", () => {
    const component = shallow(<TimeMarks videoDuration={3600} />);
    expect(component).toMatchSnapshot();
  });
});
