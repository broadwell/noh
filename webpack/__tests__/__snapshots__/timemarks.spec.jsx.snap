// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeMarks> renders as expected 1`] = `
<div
  className="time-marks"
>
  <div
    className="time"
  >
    0:00:00
  </div>
  <div
    className="time"
  >
    0:05:00
  </div>
  <div
    className="time"
  >
    0:10:00
  </div>
  <div
    className="time"
  >
    0:15:00
  </div>
  <div
    className="time"
  >
    0:20:00
  </div>
  <div
    className="time"
  >
    0:25:00
  </div>
  <div
    className="time"
  >
    0:30:00
  </div>
  <div
    className="time"
  >
    0:35:00
  </div>
  <div
    className="time"
  >
    0:40:00
  </div>
  <div
    className="time"
  >
    0:45:00
  </div>
  <div
    className="time"
  >
    0:50:00
  </div>
  <div
    className="time"
  >
    0:55:00
  </div>
  <div
    className="time"
  >
    1:00:00
  </div>
  <div
    className="time"
  >
    1:05:00
  </div>
  <div
    className="time"
  >
    1:10:00
  </div>
  <div
    className="time"
  >
    1:15:00
  </div>
</div>
`;
